src/modules/function.js

```diff
@@ -0,0 +1,111 @@
+import List from './list.js';
+
+const form = document.getElementById('list-form');
+const listContainer = document.getElementById('list-container');
+let tasks = [];
+let id = 0;
+
+// Function to save tasks to local storage
+function saveTasksToLocalStorage() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+// Function to render the list
+function renderList() {
+  listContainer.innerHTML = '';
+
+  tasks.forEach((task, index) => {
+    const checkbox = document.createElement('input');
+    const remove = document.createElement('button');
+    const strdiv = document.createElement('div');
+    const btndiv = document.createElement('div');
+
+    checkbox.type = 'checkbox';
+    checkbox.checked = task.completed;
+    const listItem = document.createElement('li');
+    const edit = document.createElement('button');
+    edit.className = 'editbtn';
+    edit.innerHTML = '<i class="fa-regular fa-pen-to-square"></i>';
+    edit.addEventListener('click', () => {
+      listItem.contentEditable = 'true';
+    });
+
+    checkbox.addEventListener('change', () => {
+      task.completed = checkbox.checked;
+      if (checkbox.checked) {
+        listItem.style.textDecoration = 'line-through';
+        listItem.style.fontStyle = 'italic';
+      } else {
+        listItem.style.textDecoration = 'none';
+        listItem.style.fontStyle = 'normal';
+      }
+      saveTasksToLocalStorage();
+    });
+
+    remove.className = 'removebtn';
+    remove.innerHTML = '<i class="fas fa-trash"></i>';
+    remove.addEventListener('click', () => {
+      tasks.splice(index, 1);
+      renderList();
+      saveTasksToLocalStorage();
+    });
+
+    listItem.addEventListener('keydown', (e) => {
+      if (e.key === 'Enter') {
+        e.preventDefault();
+        listItem.contentEditable = 'false';
+        task.description = listItem.innerText.trim();
+        saveTasksToLocalStorage();
+      }
+    });
+
+    strdiv.appendChild(checkbox);
+    strdiv.appendChild(document.createTextNode(task.description));
+    btndiv.appendChild(edit);
+    btndiv.appendChild(remove);
+    listItem.appendChild(strdiv);
+    listItem.appendChild(btndiv);
+    listContainer.appendChild(listItem);
+  });
+}
+
+// Function to handle form submission
+function handleFormSubmit() {
+  const input = document.getElementById('input-id');
+  const todo = new List(input.value, Boolean(), id);
+  tasks.push(todo);
+  id += 1;
```

Comment:
- [ ] Items can be added to the list correctly! 👌 However, _Index_ property of added item does not match the new array length. For instance, if a 5th item is added to the list, the index of that item should equal 5. Let's improve our code so that it meets this requirement! 😃 

**Mismatch in index property and array length:**
![image](https://github.com/noelfoka/to-do-list/assets/12745474/8fdaa0fe-3cc9-4ec2-b2bf-4ebbcdab29c1)